tests/RobotFramework/tests/pkcs11/private_key_storage_aws.robot

```diff
@@ -21,6 +22,55 @@ Connect to AWS Using PKCS11 Private Key
     ThinEdgeIO.Service Health Status Should Be Up    tedge-mapper-aws
     ThinEdgeIO.Bridge Should Be Up    aws
 
+Connect to AWS and Cumulocity using different keys
+    [Documentation]    Verify that we can have two different connections open at the same time to different clouds using
+    ...    different tokens/private keys because it's not very secure to use the same keys for different clouds.
+
+    Execute Command    tedge disconnect aws
+    Execute Command    tedge disconnect c8y
+
+    # setup: we have two key/cert pairs, each for one cloud
+    # backup token & cert we currently use for aws
+    Execute Command    mkdir /etc/tedge/device-certs/aws
+    Execute Command    mv /etc/tedge/device-certs/tedge-certificate.pem /etc/tedge/device-certs/aws
+
+    Execute Command    tedge config set aws.device.cert_path /etc/tedge/device-certs/aws/tedge-certificate.pem
+
+    # token created in setup by HSM init
+    Execute Command    cmd=tedge config set aws.device.key_uri pkcs11:token=tedge
+
+    # c8y setup: create a new token and cert/key and upload it
+    Execute Command    tedge cert create --device-id ${DEVICE_SN}
+    Execute Command
+    ...    cmd=sudo env C8Y_USER='${C8Y_CONFIG.username}' C8Y_PASSWORD='${C8Y_CONFIG.password}' tedge cert upload c8y
+    ...    log_output=${False}
+
+    Execute Command    softhsm2-util --init-token --free --label c8y-token --pin "123456" --so-pin "123456"
+    Execute Command
+    ...    bash -c 'softhsm2-util --import <(cat "$(tedge config get device.key_path)" && cat "$(tedge config get device.cert_path)") --token c8y-token --label c8y-key --id 01 --pin 123456'
+
+    Execute Command    mkdir /etc/tedge/device-certs/c8y
+    Execute Command    mv /etc/tedge/device-certs/tedge-certificate.pem /etc/tedge/device-certs/c8y
+    Execute Command    mv /etc/tedge/device-certs/tedge-private-key.pem /etc/tedge/device-certs/c8y
```

Comment:
We should delete the private key if it isn't being used (and should not be used).

Reply:
fixed in a8a9a113bd